x/ccv/consumer/keeper/relay.go

```diff
@@ -86,46 +85,30 @@ func (k Keeper) OnRecvVSCPacket(ctx sdk.Context, packet channeltypes.Packet, new
 // operations that resulted in validator updates included in that VSC have matured on
 // the consumer chain.
 func (k Keeper) QueueVSCMaturedPackets(ctx sdk.Context) {
-	store := ctx.KVStore(k.storeKey)
-	maturityIterator := sdk.KVStorePrefixIterator(store, []byte{types.PacketMaturityTimeBytePrefix})
-	defer maturityIterator.Close()
-
-	currentTime := uint64(ctx.BlockTime().UnixNano())
-
-	maturedVscIds := []uint64{}
-	for maturityIterator.Valid() {
-		vscId := types.IdFromPacketMaturityTimeKey(maturityIterator.Key())
-		if currentTime >= binary.BigEndian.Uint64(maturityIterator.Value()) {
-			// construct validator set change packet data
-			vscPacket := ccv.NewVSCMaturedPacketData(vscId)
-
-			// append VSCMatured packet to pending packets
-			// sending packets is attempted each EndBlock
-			// unsent packets remain in the queue until sent
-			k.AppendPendingPacket(ctx, types.ConsumerPacket{
-				Type: types.VscMaturedPacket,
-				Data: vscPacket.GetBytes(),
-			})
-
-			ctx.EventManager().EmitEvent(
-				sdk.NewEvent(
-					ccv.EventTypeVSCMatured,
-					sdk.NewAttribute(sdk.AttributeKeyModule, types.ModuleName),
-					sdk.NewAttribute(ccv.AttributeChainID, ctx.ChainID()),
-					sdk.NewAttribute(ccv.AttributeConsumerHeight, strconv.Itoa(int(ctx.BlockHeight()))),
-					sdk.NewAttribute(ccv.AttributeValSetUpdateID, strconv.Itoa(int(vscId))),
-					sdk.NewAttribute(ccv.AttributeTimestamp, strconv.Itoa(int(currentTime))),
-				),
-			)
-
-			maturedVscIds = append(maturedVscIds, vscId)
-		} else {
-			break
-		}
-		maturityIterator.Next()
-	}
+	vscIDs := k.DequeueAllMatureVSCPacketQueue(ctx)
```

Comment:
I would personally prefer that we don't delete inside `DequeueAllMatureVSCPacketQueue` and delete in `QueueVSCMaturedPackets`.

The implementation reads in a weird way. This function seems as if it is simply providing the caller with a list of vscIDs.

Reply:
I would name `DequeueAllMatureVSCPacketQueue` something like `ConsumeMatureVSCPacketQueue` because that makes it more clear that it deletes stuff.

Reply:
I prefer to keep it consistent with the `DequeueAllMatureUBDQueue` from the SDK.